### PR TITLE
Upgrade to Astro v3

### PR DESCRIPTION
**Pull request by lukewarlow**
This upgrades the Astro framework to v3. This should lead to a much faster dev environment along with unlocking niceties such as built in view transitions should we want to use them (see #797 )

See https://docs.astro.build/en/guides/upgrade-to/v2/ for v1 to v2

See https://docs.astro.build/en/guides/upgrade-to/v3/ for v2 to v3

The only major thing this does is increase the minimum node requirement to node 18.

It also might change the shape of the build output not sure on that one.

### CODE REVIEW

File: site/src/pages/404.jsx [hunk 1]
```diff
@@ -1,14 +0,0 @@
-import React from 'react'
```
Comment (andrico1234) [inline site/src/pages/404.jsx:1]: Does Astro add its own default 404 page ?

File: site/src/pages/contribute.astro [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+return Astro.redirect('/get-involved');
```
Comment (andrico1234) [inline site/src/pages/contribute.astro:2]: Much nicer! Great spot

**Comment by andrico1234** [comment]
Code wise, everything looks great! I'll give the preview deploy a test over the next few days. Thanks for this!

File: site/src/pages/404.jsx (hunk 1 above)
Comment (lukewarlow) [reply site/src/pages/404.jsx:1]: Astro no longer supports JSX components. And I checked and netlify provided its own 404 so I don't believe this one is ever seen

**Review by andrico1234 — APPROVED** [review]

**Comment by andrico1234** [comment]
Thanks for this! I've approved. It looks like you need a Code Owner to approve as well

**Comment by mfreed7** [comment]
> Thanks for this! I've approved. It looks like you need a Code Owner to approve as well

I'll approve based on this! (I'm no Astro expert.)

**Review by mfreed7 — APPROVED** [review]

**Pull request merged**